### DISCUSSION TITLE
How rigid bodies are given parameters

### DISCUSSION THREAD

**Discussion by guwulang** [Anura3D/Anura3D_OpenSource#225] [General]
Dear teachers,
    I would like to ask how to set the parameters under the rigid body module in the setting material properties (version is Anura3D2022, 14.01), 0 is to limit its movement in a certain direction, and 1 is not limited. I set all 3 directions to 0, the whole section will sink, how can I improve it? Looking forward to hearing from you.
![V %Q{GJ@QGF5{JQ($(UHZ70](https://github.com/Anura3D/Anura3D_OpenSource/assets/167985770/7ddb39b5-eb39-44e3-8609-b4ab476133bc)
![3G${DJ5KZVB}G%O Y4O5 DC](https://github.com/Anura3D/Anura3D_OpenSource/assets/167985770/50e28c83-8e03-432d-972e-c857e825cf3e)
![X7A4Z TK5C5MUODM29S}RL2](https://github.com/Anura3D/Anura3D_OpenSource/assets/167985770/089f0251-9c11-4809-a126-773a90bd33c9)
![LWI48ZKC}MQQ{61U07WLV H](https://github.com/Anura3D/Anura3D_OpenSource/assets/167985770/0c21354f-aba4-4994-ba92-ea87d56b5eea)

**Comment by FrancescaCeccato** [comment]
It is the opposite. 1 means that the direction is constraint (cannot move in that direction) and 0 means it is not constraint (it can move in that direction).
Please remeber the limitation of rigid body, i.e. can be applied only to one body in the system, only works together with the
contact algorithm, only moves in one direction, it can’t rotate (see manual, sec. 1.3).

**Reply by CPLA81192** [reply]
When I use rigid bodies as a shaking table to simulate earthquake loads, I noticed that the rigid bodies have vertical displacements and show vertical compression. What could be the reason for this?
![图片1](https://github.com/user-attachments/assets/67d62271-df38-4ae1-b069-6cf875d72202)
